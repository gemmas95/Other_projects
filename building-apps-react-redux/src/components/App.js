/* eslint-disable react/jsx-no-duplicate-props */
import React from "react";
import { Route } from "react-router-dom";
import HomePage from "./home/HomePage";
import AboutPage from "./about/AboutPage";

function App() {
  return (
    <div className="container-fluid">
      <Route exact component path="/" component={HomePage} />
      <Route component path="/about" component={AboutPage} />
    </div>
  );
}

export default App;
