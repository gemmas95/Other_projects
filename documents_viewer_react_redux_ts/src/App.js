import { Switch, Route } from "react-router-dom";
import styled from "styled-components";
import { ToastContainer } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";

// Components
import Navbar from "./components/common/Navbar";
import Home from "./containers/homePage/Home";
import DocumentsPage from "./containers/DocumentsPage";
import DocumentInfo from "./containers/documentInfoPage/DocumentInfo";
import ManageNewDocumentsPage from "./containers/ManageNewDocumentsPage";
import PageNotFound from "./components/common/PageNotFound";

const GeneralStyles = styled.div`
  padding: 10px 30px;
  text-align: center;

  @media (min-width: 560px) {
    padding: 30px 100px;
    text-align: left;
  }

  @media (min-width: 980px) {
    padding: 30px 300px;
  }
`;

function App() {
  return (
    <GeneralStyles>
      <Navbar />
      <Switch>
        <Route path="/documents/:documentId" exact component={DocumentInfo} />
        <Route path="/documents" component={DocumentsPage} />
        <Route path="/newdocument" component={ManageNewDocumentsPage} />
        <Route path="/" exact component={Home} />
        <Route component={PageNotFound} />
      </Switch>
      <ToastContainer
        position="top-right"
        hideProgressBar={false}
        newestOnTop={false}
        closeOnClick
        rtl={false}
        pauseOnFocusLoss
        draggable
        pauseOnHover
      />
    </GeneralStyles>
  );
}

export default App;
