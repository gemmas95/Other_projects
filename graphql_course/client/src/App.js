import React from "react";
// Components
import BookList from "./components/BookList";
import AddBook from "./components/AddBook";

function App() {
  return (
    <div id="main">
      <h1>Ninja's Reading List!</h1>
      <BookList />
      <AddBook />
    </div>
  );
}

export default App;
