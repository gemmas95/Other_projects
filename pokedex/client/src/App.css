.App {
  text-align: center;
}

.pokemons__list {
  padding: 0px;
  display: flex;
  flex-wrap: wrap;
  margin: 100px 27em;
  justify-content: center;
}

.pokemons__list li {
  list-style: none;
  margin: 10px;
}

.currentLink {
  background: rgb(216, 216, 216);
}
.currentLink > a {
  text-decoration: none;
  color: gray;
}
/* Working on it */
.pokemons__list li p:first-child {
  color: red;
  background: white;
}

.pagination {
  display: flex;
  padding: 0;
  background: #eee;
  border: 1px solid gray;
  border-radius: 4px;
  justify-content: center;
}

.pagination li {
  list-style: none;
  padding: 10px;
  border: 1px solid gray;
}
