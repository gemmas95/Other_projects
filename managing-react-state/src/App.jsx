import React from "react";
import "./App.css";
import Footer from "./Footer";
import Header from "./Header";
import Products from "./Products";
import { Routes, Route } from "react-router-dom";
import Detail from "./Detail.class";
import Cart from "./Cart";
import Checkout from "./Checkout.class";
import { useCart } from "./cartContext";

export default function App() {
  const { dispatch } = useCart();
  /* MOVED IN CARTREDUCER
     function addToCart(id, sku) {
    setCart((items) => {
      const itemInCart = items.find((i) => i.sku === sku);
      if (itemInCart) {
        // Return new array with the matching item replaced
        return items.map((i) =>
          i.sku === sku ? { ...i, quantity: i.quantity + 1 } : i
        );
      } else {
        // Return new array with the new item appended
        return [...items, { id, sku, quantity: 1 }];
      }
    });
  }

  function updateQuantity(sku, quantity) {
    setCart((items) => {
      return quantity === 0
        ? items.filter((i) => i.sku !== sku)
        : items.map((i) => (i.sku === sku ? { ...i, quantity } : i));
    });
  }

  function emptyCart() {
    setCart([]);
  } */

  return (
    // We are saying these values can be consume using the CartContext
    <>
      <div className="content">
        <Header />
        <main>
          <Routes>
            <Route path="/" element={<h1>Welcome to Carved Rock Fitness</h1>} />
            <Route path="/:category" element={<Products />} />
            <Route path="/:category/:id" element={<Detail />} />
            <Route path="/cart" element={<Cart />} />
            <Route
              path="/checkout"
              element={<Checkout dispatch={dispatch} />}
            />
          </Routes>
        </main>
      </div>
      <Footer />
    </>
  );
}
