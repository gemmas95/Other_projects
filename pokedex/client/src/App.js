import "./App.css";
import CardList from "./components/CardList";
import Header from "./components/Header";

function App() {
  return (
    <div className="App">
      <Header />
      <CardList />
    </div>
  );
}

export default App;
