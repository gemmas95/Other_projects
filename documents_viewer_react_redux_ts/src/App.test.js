import { render } from "@testing-library/react";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import App from "./App";

describe("App", () => {
  it("should match snapshot", () => {
    const AppTree = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    expect(AppTree).toMatchSnapshot();
  });
});
