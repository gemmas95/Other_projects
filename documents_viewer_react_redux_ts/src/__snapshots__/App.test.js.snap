// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-dlfnbm faFStx"
      >
        <nav>
          <a
            aria-current="page"
            class="active"
            href="/"
            style="color: rgb(35, 75, 85);"
          >
            Home
          </a>
           
           | 
          <a
            href="/documents"
          >
            Documents
          </a>
        </nav>
        <h1>
          Welcome to Documents viewer!
        </h1>
        <p>
          This applications is made for Signaturit Jr Frontend Developer job application.
        </p>
        <div
          class="Toastify"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-dlfnbm faFStx"
    >
      <nav>
        <a
          aria-current="page"
          class="active"
          href="/"
          style="color: rgb(35, 75, 85);"
        >
          Home
        </a>
         
         | 
        <a
          href="/documents"
        >
          Documents
        </a>
      </nav>
      <h1>
        Welcome to Documents viewer!
      </h1>
      <p>
        This applications is made for Signaturit Jr Frontend Developer job application.
      </p>
      <div
        class="Toastify"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
